addons/web/static/src/core/scroller_service.js

```diff
@@ -35,12 +35,10 @@ export const scrollerService = {
                         ev.preventDefault(); // single hash in href are just a way to activate A-tags node
                         return;
                     }
-                    let matchingEl = null;
-                    try {
-                        matchingEl = document.querySelector(`.o_content #${href.value.substr(1)}`);
-                    } catch {
-                        // Invalid selector: not an anchor anyway
-                    }
+                    const selector = href.value.substr(1);
+                    const content = document.querySelector(".o_content");
+                    const el = selector.match(/\s/g) ? null : document.getElementById(selector);
```

Comment:
the spaces do not appear to make crash document.getElementById

Reply:
Exact, indeed. But as it a requirement of HTML5, it's maybe a good idea to check as you proposed.